### ISSUE TITLE
[Helm 3] I have a question about 'switch ... case ' syntax

### ISSUE THREAD

**Issue by fengxsong** [helm/helm#6440]
- https://github.com/helm/helm/blob/5cb923eecbe80d1ad76399aee234717c11931d9a/pkg/kube/wait.go#L67
- https://github.com/helm/helm/blob/5cb923eecbe80d1ad76399aee234717c11931d9a/pkg/kube/wait.go#L80
- https://github.com/helm/helm/blob/5cb923eecbe80d1ad76399aee234717c11931d9a/pkg/kube/wait.go#L93
- https://github.com/helm/helm/blob/5cb923eecbe80d1ad76399aee234717c11931d9a/pkg/kube/wait.go#L106

Wondering why not use syntax as follow

```go
...
            switch value := AsVersioned(v).(type) {
            case *corev1.Pod:
              ...
            case *appsv1.Deployment, *appsv1beta1.Deployment, *appsv1beta2.Deployment, *extensionsv1beta1.Deployment:
              ...
              ...
            }
```

**Comment by hickeyma** [comment]
@fengxsong That is a good suggestion for refactor. Do you have to push a PR for this?

**Comment by bacongobbler** [comment]
Closed in #6608
